### DISCUSSION TITLE
The case for contravariant template types – Blog

### DISCUSSION THREAD

**Discussion by giscus** [jiripudil/jiripudil.cz#113] [Blog discussions]
# The case for contravariant template types – Blog

A full-stack web developer from Brno

https://jiripudil.cz/blog/contravariant-template-types

**Comment by voku** [comment]
Thank you for your contribution and for making the entire PHP world better with your work.

**Comment by jkuchar** [comment]
Thank you for your contribution. The contravariance & generics are a huge step toward making PHP a mature and usable language suitable for long-term projects. ❤️
